### ISSUE TITLE
Context menu scrolling arrows (an endless problem of CR2023)

### ISSUE THREAD

**Issue by firegold763** [Alex313031/thorium#769]
**System Details**
 - OS: Windows 10
 - Thorium Version: 126.0.6478.231

**Problem**
The good old scaling/context menu problem that Chrome has since they released CR2023, in laptop 1080p/125% scaling it's unusable since it's too big and arrows for scrolling came up.

**Screenshots**
![Screenshot (79)](https://github.com/user-attachments/assets/82ee9b57-9acb-4a92-891e-1dacb74f2826)

**Additional Notes**
I expected this to happen hoping that maybe with Thorium it could be different, maybe trying to mantain the old UI, instead no it seems that nothing could be done unless Google themselves do something but apparently they don't since it's MONTHS that is like this.

**Comment by People-11** [comment]
I will stay at M124. The dev seems don't have the ability to really touch the code. Just adjusting compile flags, merging upstreams, mitigating patches, that's all.

Maybe they have the ability, but they really don't care. You have this bug at M126, and I have a bug that always shows a new icon in context menu. They might found the problems in testing but don't care to fix them.

They just close issues about performance because Chrome started using Maglev compiler so Thorium's performance lead over feature was no longer obvious. They also ignored all the "Hide tab search button" issues and provide a position flag without hiding option. They just said "Refresh 2023 is the default UI" because it's Google completed removed old UI and they didn't do any. They just refuse to implement simple features, for their own reasons, and haven't considered making them an option to really make us feel good about using it.

Some might say "don't talk, just fork yourself, and we will appreciate your PR". So why do Bug Report, Feature Request and General Feedback exist? "We want your ideas and findings but we won't make them come true"?

**Comment by firegold763** [comment]
> I will stay at M124. The dev seems don't have the ability to really touch the code. Just adjusting compile flags, merging upstreams, mitigating patches, that's all.
> 
> Maybe they have the ability, but they really don't care. You have this bug at M126, and I have a bug that always shows a new icon in context menu. They might found the problems in testing but don't care to fix them.
> 
> They just close issues about performance because Chrome started using Maglev compiler so Thorium's performance lead over feature was no longer obvious. They also ignored all the "Hide tab search button" issues and provide a position flag without hiding option. They just said "Refresh 2023 is the default UI" because it's Google completed removed old UI and they didn't do any. They just refuse to implement simple features, for their own reasons, and haven't considered making them an option to really make us feel good about using it.
> 
> Some might say "don't talk, just fork yourself, and we will appreciate your PR". So why do Bug Report, Feature Request and General Feedback exist? "We want your ideas and findings but we won't make them come true"?

Yeah honestly I agree, the dev always said he hated the new UI and would have done all he could to try and maintain it, yet instead we are here with the new flawed UI that Google made only for 4K desktop screens with 100% scaling.

At this point there's no difference at using basic Chrome or Thorium, maybe just Manifest V2? But that's all.

**Comment by gz83** [comment]
If you are not happy with Thorium Browser and cannot see the hard work of Alex, then I can only advise you not to use it.

We are still trying to perfect the interface of the browser, but the whole process is a best-effort.

**Comment by Alex313031** [comment]
@firegold763 @People-11 Already working on fixing the ugly Cr23 UI.

It restores the tab height, tab separators, and menus are square instead of round.
More work in progress now to put it all properly behind a flag.

I can also try to make it decrease the padding in menus to the way it was before.

See commit here > https://github.com/Alex313031/thorium/commit/07bd646449bedc683b833e2c931747ad4aee27ed

Thorium normally would have had the new ugly UI in M122. I staved it off for as long as possible by disabling internal flags for the 2023 UI. But now they are removing the code since M125, so I'm having to come up with my own solution.

Modifying the UI isn't the easiest thing, mainly just because of trying to find where in the code is what you are trying to modify. Once I found those lines for example, that have static integers for pixel spacing on certain UI elements, I could make it where the values are what they were in the pre-2023 UI, and put that behind a flag.
And I have the "ability to touch the code", I'm just not very good but competent enough to do it after lots of trial and error.  Sometimes I decide not to add something even though it was asked in the community (even multiple times).

**Comment by firegold763** [comment]
> @firegold763 @People-11 Already working on fixing the ugly Cr23 UI.
> 
> It restores the tab height, tab separators, and menus are square instead of round. More work in progress now to put it all properly behind a flag.
> 
> I can also try to make it decrease the padding in menus to the way it was before.
> 
> See commit here > [07bd646](https://github.com/Alex313031/thorium/commit/07bd646449bedc683b833e2c931747ad4aee27ed)
> 
> Thorium normally would have had the new ugly UI in M122. I staved it off for as long as possible by disabling internal flags for the 2023 UI. But now they are removing the code since M125, so I'm having to come up with my own solution.
> 
> Modifying the UI isn't the easiest thing, mainly just because of trying to find where in the code is what you are trying to modify. Once I found those lines for example, that have static integers for pixel spacing on certain UI elements, I could make it where the values are what they were in the pre-2023 UI, and put that behind a flag. And I have the "ability to touch the code", I'm just not very good but competent enough to do it after lots of trial and error. Sometimes I decide not to add something even though it was asked in the community (even multiple times).
> 
> Also @gz83 these comments didn't need to be hidden. While they were critical I don't think it was something worth hiding.

It's more Google's fault clearly, they had tons and tons of criticism, tons of people that called them out for the scrolling menu problem, because they didn't think about screens that either ain't 4K or either don't have 100% scaling but a higher one. It's since October 2023 that the problem surfaced, but they did absolutely nothing, that's why I've always downgraded and changed browsers here and there. I've seen Brave fixed it by heavily modifying the menus, reducing the padding, and stuff like that.

But god they just fucked up everything on laptops or lower res screens, it's so annoying.

**Comment by gz83** [comment]
I agree with you and I know that there are many negative comments about CR23 style. But even so, I think users should respect developers as much as possible instead of complaining all the time.

@Alex313031

**Comment by firegold763** [comment]
> I agree with you and I know that there are many negative comments about CR23 style. But even so, I think users should respect developers as much as possible instead of complaining all the time.
> 
> @Alex313031

The other user was harsher, talk to him, not to me. Plus, if you can't take complaints, then don't open a "Bug Report" thread. Take the critics and make something out of them, you just seem offended by them.

**Comment by Zelassin** [comment]
thank you for at least trying to fix the ui. If it's ultimately impossible to do so, how bad for safety is it to just stay on the 124 version of the browser?

**Comment by Amanoki** [comment]
Opening feedbacks only means that devs may possibly consider those opinions and ideas and implement them when they feel like it and are able to. Devs are not obligated to satisfy anyone's requests. It's understandable even if devs not doing them for whatever reason and consideration.

Leaving the comment as I share OP's wish for the old UI to be brought back, glad to know it's in progress. Personally the spacing in the menus is the most crucial, it's the direct cause of Chrome becoming unusable for me anymore, would love to see it to get addressed eventually. Regardless, just do what you can and take your time. I appreciate all the hard work Alex and the contributors have put into Thorium.

**Comment by People-11** [comment]
@Alex313031 Then could you consider [this](https://github.com/Alex313031/thorium/issues/590)? I don't want to hear from gz83. Because URLEncode or Decode is not about `different language specifications and other compatibility factors`, it just convert strings to ASCII and Hex them.

**Comment by firegold763** [comment]
> Opening feedbacks only means that devs may possibly consider those opinions and ideas and implement them when they feel like it and are able to. Devs are not obligated to satisfy anyone's requests. It's understandable even if devs not doing them for whatever reason and consideration.
> 
> Leaving the comment as I share OP's wish for the old UI to be brought back, glad to know it's in progress. Personally the spacing in the menus is the most crucial, it's the direct cause of Chrome becoming unusable for me anymore, would love to see it to get addressed eventually. Regardless, just do what you can and take your time. I appreciate all the hard work Alex and the contributors have put into Thorium.

I agree, but this is literally a bug. It's not a feature request or not even an opinion or suggestion.

**Comment by gz83** [comment]
I have given you enough opportunities to discuss, but there are still irrelevant content or comments that are disrespectful to team members. I will continue to edit or limit such comments.

Whether to modify or add certain features is not decided by me, and some of my comments are neutral. At the same time, modifying or adding certain features will have a significant impact on the user group, and it is something that must be treated with caution.

**Comment by Amanoki** [comment]
> I agree, but this is literally a bug. It's not a feature request or not even an opinion or suggestion.

CR2023 and the scrolling context menus were brought in by Chrome as a feature _on purpose_ as you already know, you could probably call it bug to Google and tell them to fix it.
For Thorium this actually required extra new work to rebuild the old UI to satisfy users' preferences for the old UI or the need for compact menus, which is essentially a feature request.

**Comment by firegold763** [comment]
> > I agree, but this is literally a bug. It's not a feature request or not even an opinion or suggestion.
> 
> CR2023 and the scrolling context menus were brought in by Chrome as a ~feature on purpose~ as you already know, you could probably call it bug to Google and tell them to fix it. (Edited: it's indeed an issue finally acknowledged by Google just days ago, there's still hope for Google to fix it) For Thorium this actually requires extra new work to rebuild the UI to satisfy users' preferences for the old UI or the need for compact menus, which is essentially a feature request.

At least it seems they're doing something, maybe it was a feature at first, but then it became an UI Regression (at least on 1080p/125% scaling screens) and then a consequentially a sort of bug. I just hope it gets fixed, since for the rest I don't even mind too much the new UI, but the context menus are annoying.

**Comment by xFiskx** [comment]
Chrome 129 has a new compact function built in

flag: Compact mode for the browser - Prototype

**Comment by Amanoki** [comment]
Compact mode doesn't seem to touch the menus. Another problem with the context menu besides padding is that its height is limited by the mouse click position, instead of utilizing the full height of the browser. Someone there seems to be working on the issue unfortunately no new news since August 1. https://chromium-review.googlesource.com/c/chromium/src/+/5547329

**Comment by WmeLuna** [comment]
my current workaround, until thorium is relased with the fixes, is using [Chrome UI Tweaks](https://windhawk.net/mods/chrome-ui-tweaks) with thorium.exe on the custom inclusion list, not perfect but it means i can use 126 without the annoying context menu :P

**Comment by firegold763** [comment]
> @Amanoki @WmeLuna @Zelassin @firegold763 @People-11 @xFiskx New beta "Th24" release that fixes this and more > https://github.com/Alex313031/thorium/releases/tag/M126.0.6478.246

Thanks!

Appreciate your hard work, I'll wait for the Windows version to come out and try it out.

**Comment by Amanoki** [comment]
Same, can't wait for the Windows release! Thanks for the update!

**Comment by Alex313031** [comment]
@Amanoki @WmeLuna @Zelassin @firegold763 @People-11 @xFiskx Making a Window build now, which has even more fixes on top of the Linux build.

**Comment by firegold763** [comment]
> @Amanoki @WmeLuna @Zelassin @firegold763 @People-11 @xFiskx Making a Window build now, which has even more fixes on top of the Linux build.

Good to know, looking forward to it

**Comment by Amanoki** [comment]
@Alex313031 I've been playing with Th24 Beta for a while, a few things I've noticed so far:
1. This should be related to this issue so I'll put the details here. The context menu padding in Th24 seem still slightly larger than the old UI. On 124, my context menu items was able to fit exactly within my screen height, whereas on Th24 Beta, about 5~6 items become hidden under the scroll. It's not really a problem, however, as the difference is minor, so I am just mentioning this for your consideraction. It would be nice if the padding becomes a bit futher smaller (same as the old UI). The visual change wouldn't be noticeable comparing to the current value, but would allow more items to be displayed.

![image](https://github.com/user-attachments/assets/8912fad3-07d7-48f3-85ad-f4ace98d20b5)
(White is 124, black is Th24)

2. When the browser is at a non-maximized (restored) state, the top edge of window becomes uninteractable, unable to resize the window by dragging it.
3. Unable to activate a tab by clicking on the top edge of tab. (Same as [this](https://github.com/Alex313031/Thorium-Win/issues/248#issuecomment-2307427140))
4. Weird white line appears under the bookmark bar when custom themes installed. (Made a seperate issue [here](https://github.com/Alex313031/Thorium-Win/issues/263))

Nevertheless, I like Th24 very much! Most of the parts (that I care about) are very nicely restored to the previous experience. Guess I can finally move from 124 once the stable version comes out. Thank you for your impressing work!

**Comment by Eff0rtly** [comment]
@Alex313031  To be honest, you should look at the source code of this ui tweaks. You don't have to restore the old UI but should tweak it like this or brave. https://windhawk.net/mods/chrome-ui-tweaks

**Comment by Alex313031** [comment]
@Amanoki Added to my TODO to fix.
